website_portal_contract/tests/test_controller.py

```diff
@@ -19,3 +19,11 @@ def test_portal_contract_view_tour(self):
             ready="%s.tours.test_contract_view.ready" % tour,
             login='portal',
         )
+
+    def test_template_view(self):
+        """ It should respond with 200 status """
+        response = self.url_open('/contract/template/1')
```

Comment:
You're not triggering the controller, which means you're likely throwing in a bad ID. Are you sure the template ID is one? Why not use an `env.ref().id` instead

Reply:
Nope. You're not logged in, so it's sending you to the login page.

```
self.authenticate('admin', 'admin')
self.url_open
```

Reply:
oh i see interesting. The id in this case is definitely 1, but ye better idea to do ref.

I didn't realize you had to explicitly authenticate in this case, which solves my confusion as to why the method wasn't being covered.